### PR TITLE
[release] v6.18.1

### PR DESCRIPTION
**Pull request by michelengelen**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).
---
The following steps must be proposed as a pull request.

- [x] Compare the last tag with the branch upon which you want to release (`next` for the alpha / beta releases and `master` for the current stable version).
- [x] Clean the generated changelog, to match the format of [https://github.com/mui/mui-x/releases](https://github.com/mui/mui-x/releases).
- [x] Update the root `package.json`'s version
- [x] Update the versions of the other `package.json` files and of the dependencies with `yarn release:version` (`yarn release:version prerelease` for alpha / beta releases).
- [x] Open PR with changes and wait for review and green CI.
- [x] Merge PR once CI is green, and it has been approved.

### Release the packages

- [x] Checkout the last version of the working branch
- [x] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Push the newly created tag: `yarn release:tag`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,59 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.18.1
```
Comment (LukasTy) [inline CHANGELOG.md:6]:
```suggestion
## 6.18.1

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,59 @@
...
+We'd like to offer a big thanks to the 9 contributors who made this release possible. Here are some highlights ✨:
+
+- ✨ Multi input date time field section selection
+- 📈 ChartsReferenceLine component is now available
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+
+### Data Grid
+
+#### `@mui/x-data-grid@__VERSION__`
```
Comment (LukasTy) [inline CHANGELOG.md]: You probably forgot to replace all `__VERSION__` instances 🙈

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,59 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.18.1
+_Nov 9, 2023_
+
+We'd like to offer a big thanks to the 9 contributors who made this release possible. Here are some highlights ✨:
+
+- ✨ Multi input date time field section selection
```
Comment (LukasTy) [inline CHANGELOG.md:12]: It's just a tiny bugfix. 🙈 
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,59 @@
...
+- 📈 ChartsReferenceLine component is now available
```
Comment (LukasTy) [inline CHANGELOG.md:13]:
```suggestion
- 📈 <ChartsReferenceLine /> component is now available
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,59 @@
...
+Same changes as in `@mui/x-date-pickers@__VERSION__`.
+
+### Charts / `@mui/x-charts@__VERSION__`
+
+- [charts] Add <ChartsReferenceLine /> component (#10597) @wascou
+- [charts] Improve properties JSDoc (#10931) @alexfauquette
+
+### Docs
+
+- [docs] Fix charts docs as stable (#10888) @alexfauquette
```
Comment (alexfauquette) [inline CHANGELOG.md:53]:
```suggestion
- [docs] Fix charts docs as stable (#10888) @alexfauquette
- [docs] Document how to hide the legend (#10954) @alexfauquette
```

**Review by LukasTy — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (michelengelen) [reply CHANGELOG.md]: you know better than me what is worth mentioning here! :D

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,60 @@
...
+
+#### `@mui/x-data-grid@6.18.1`
+
+- [DataGrid] Fix cell value type in quick filtering v7 (#10884) @cherniavskii
+- [DataGrid] Fix keyboard navigation for actions cell with disabled buttons (#10947) @michelengelen
+- [DataGrid] Fix undefined slot values (#10934) @romgrk
+
+#### `@mui/x-data-grid-pro@6.18.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-data-grid@__VERSION__`, plus:
```
Comment (alexfauquette) [inline CHANGELOG.md:26]:
```suggestion
Same changes as in `@mui/x-data-grid@6.18.1`, plus:
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,60 @@
...
+
+- [DataGridPro] Add data source interface and basic documentation (#10543) @MBilalShafi
+
+#### `@mui/x-data-grid-premium@6.18.1` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')
+
+Same changes as in `@mui/x-data-grid-pro@__VERSION__`.
```
Comment (alexfauquette) [inline CHANGELOG.md:32]:
```suggestion
Same changes as in `@mui/x-data-grid-pro@6.18.1`.
```

**Review by alexfauquette — APPROVED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,62 @@
...
+
+Same changes as in `@mui/x-data-grid@6.18.1`, plus:
+
+- [DataGridPro] Add data source interface and basic documentation (#10543) @MBilalShafi
+
+#### `@mui/x-data-grid-premium@6.18.1` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')
+
+Same changes as in `@mui/x-data-grid-pro@6.18.1`, plus:
+
+- [DataGridPremium] Render aggregation label when `renderHeader` is used (#10961) @cherniavskii
```
Comment (michelengelen) [inline CHANGELOG.md:34]: @cherniavskii should we add this to the highlights?

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,62 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.18.1
+
+_Nov 9, 2023_
+
+We'd like to offer a big thanks to the 9 contributors who made this release possible. Here are some highlights ✨:
+
+- 📈 <ChartsReferenceLine /> component is now available
```
Comment (cherniavskii) [inline CHANGELOG.md:12]:
```suggestion
- 📈 `<ChartsReferenceLine />` component is now available
```

File: CHANGELOG.md (hunk 8 above)
Comment (cherniavskii) [reply CHANGELOG.md:34]: Seems too small for a highlight, but I'm not opposed to it 🙂

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,62 @@
...
+- [fields] Fix multi input date time field section selection (#10915) @noraleonte
+- [pickers] Always use up-to-date `defaultView` (#10889) @LukasTy
+
+#### `@mui/x-date-pickers-pro@6.18.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-date-pickers@6.18.1`.
+
+### Charts / `@mui/x-charts@6.18.1`
+
+- [charts] Add <ChartsReferenceLine /> component (#10597) @wascou
```
Comment (cherniavskii) [inline CHANGELOG.md:49]:
```suggestion
- [charts] Add `<ChartsReferenceLine />` component (#10597) @wascou
```

**Review by cherniavskii — APPROVED** [review]

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,6 +3,63 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.18.1
+
+_Nov 9, 2023_
+
+We'd like to offer a big thanks to the 9 contributors who made this release possible. Here are some highlights ✨:
+
+- ✨ Render aggregation label when `renderHeader` is used
```
Comment (michelengelen) [inline CHANGELOG.md:12]: @cherniavskii @MBilalShafi last review before merge? :P :shipit:

Comment (MBilalShafi) [inline CHANGELOG.md:12]:
```suggestion
- ✨ Fix aggregation label not showing when `renderHeader` is used (#10961) @cherniavskii
- 📘 Server side data source [early documentation](https://mui.com/x/react-data-grid/server-side-data/) published
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -3,6 +3,63 @@
...
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+
+### Data Grid
+
+#### `@mui/x-data-grid@6.18.1`
+
+- [DataGrid] Fix cell value type in quick filtering v7 (#10884) @cherniavskii
+- [DataGrid] Fix keyboard navigation for actions cell with disabled buttons (#10947) @michelengelen
+- [DataGrid] Fix undefined slot values (#10934) @romgrk
```
Comment (MBilalShafi) [inline CHANGELOG.md:23]:
```suggestion
- [DataGrid] Fix `undefined` slot values (#10934) @romgrk
```

**Review by MBilalShafi — APPROVED** [review]

**Review by flaviendelangle — APPROVED** [review]

**Pull request merged**